From fd8729b31294e8fc8d0eaccace1d1ac0c0c42049 Mon Sep 17 00:00:00 2001
From: Jason Andryuk <jandryuk@gmail.com>
Date: Fri, 20 Sep 2019 09:45:57 -0400
Subject: [PATCH 07/26] oxt: flask: introduce argo_comms

The argo_comms interface lets two domains talk argo to each other over
single source rings.  wildcard rings are not included and they require
explicit granting of that permission for the side(s) registering the
ring.

Signed-off-by: Jason Andryuk <jandryuk@gmail.com>
---
 tools/flask/policy/modules/xen.if | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/tools/flask/policy/modules/xen.if b/tools/flask/policy/modules/xen.if
index e05e07e0e6..db4a6dc892 100644
--- a/tools/flask/policy/modules/xen.if
+++ b/tools/flask/policy/modules/xen.if
@@ -211,3 +211,14 @@ define(`admin_device', `
 define(`delegate_devices', `
     allow $1 $2:resource { add remove };
 ')
+
+# argo_comms(domA, domB)
+#   Allow two domains to talk argo to each other
+define(`argo_comms', `
+    allow $1 xen_t:argo enable;
+    allow $2 xen_t:argo enable;
+    allow $1 $2:argo register_single_source;
+    allow $2 $1:argo register_single_source;
+    allow $1 $2:argo send;
+    allow $2 $1:argo send;
+')
-- 
2.21.0

